api/id/id.go

```diff
@@ -240,3 +241,42 @@ func (id *Id) UnmarshalText(v []byte) error {
 
 	return nil
 }
+
+// reverse encoding useful for sorting, descending
+var rEncoding = reverseString(Encoding)
```

Comment:
super minor, but IMHO this should just be a const like Encoding and we can remove the reverseString() below.

Reply:
Saw the TODO notes too, we'll fixup these later...